src/wasm/wasm-emscripten.cpp

```diff
@@ -742,6 +751,26 @@ Const* AsmConstWalker::resolveConstAddr(Expression* arg, const Name& target) {
   return arg->cast<Const>();
 }
 
+// Given an expression that's either (global.get $name) or (local.get $var)
+// where the most recent set on $var is (local.set $var (global.get $name)),
+// return the name.
+Name* AsmConstWalker::resolveGlobalGet(Expression* arg, const Name& target) {
+  while (!arg->dynCast<GlobalGet>()) {
+    if (auto* get = arg->dynCast<LocalGet>()) {
```

Comment:
This control flow is a bit screwy. We have a convention of dynCasting a particular node into a few different types in an if/elseif chain, but this is very explicitly looking at only LocalGet+GlobalGet, so that idiom isn't necessary.

How about:
```c++
auto* get = arg->dynCast<LocalGet>();
if (!get) {
  return nullptr;
}
auto* set = sets[get->index];
if (!set) {
  return nullptr;
}
assert(set->index == get->index);
arg = set->value;
```

Reply:
Done.